Extension/src/LanguageServer/ui.ts

```diff
@@ -58,288 +58,266 @@ const commandArguments: string[] = []; // We report the sender of the command
 
     constructor() {
-        const configTooltip: string = localize("c.cpp.configuration.tooltip", "C/C++ Configuration");
-        this.configStatusBarItem = vscode.window.createStatusBarItem("c.cpp.configuration.tooltip", vscode.StatusBarAlignment.Right, 0);
-        this.configStatusBarItem.name = configTooltip;
-        this.configStatusBarItem.command = {
-            command: "C_Cpp.ConfigurationSelect",
-            title: configTooltip,
-            arguments: commandArguments
-        };
-        this.configStatusBarItem.tooltip = configTooltip;
-        this.ShowConfiguration = true;
+        this.intelliSenseStatusItem = this.createIntelliSenseStatusItem();
+        this.tagParseStatusItem = this.createTagParseStatusItem();
+        this.codeAnalysisStatusItem = this.createCodeAnalysisStatusItem();
 
-        this.referencesStatusBarItem = vscode.window.createStatusBarItem(`c.cpp.references.statusbar`, vscode.StatusBarAlignment.Right, 901);
-        this.referencesStatusBarItem.name = localize("c.cpp.references.statusbar", "C/C++ References Status");
-        this.referencesStatusBarItem.tooltip = "";
-        this.referencesStatusBarItem.command = {
-            command: "C_Cpp.ShowReferencesProgress",
-            title: this.referencesStatusBarItem.name,
-            arguments: commandArguments
-        };
+        this.referencesStatusBarItem = this.createReferencesStatusBarItem();
         this.ShowReferencesIcon = false;
 
-        this.configureIntelliSenseStatusItem = vscode.window.createStatusBarItem(`c.cpp.configureIntelliSenseStatus.statusbar`, vscode.StatusBarAlignment.Right, 0);
-        this.configureIntelliSenseStatusItem.name = this.cppConfigureIntelliSenseText;
-        this.configureIntelliSenseStatusItem.tooltip = this.cppConfigureIntelliSenseText;
-        this.configureIntelliSenseStatusItem.text = `$(warning) ${this.configureIntelliSenseText}`;
-        this.configureIntelliSenseStatusItem.backgroundColor = new vscode.ThemeColor('statusBarItem.warningBackground');
-        this.configureIntelliSenseStatusItem.command = {
-            command: "C_Cpp.SelectIntelliSenseConfiguration",
-            title: this.configureIntelliSenseStatusItem.name,
-            arguments: ['statusBar']
-        };
+        this.configurationStatusBarItem = this.createConfigurationStatusBarItem();
+        this.ShowConfiguration = true;
+
+        this.configureIntelliSenseStatusBarItem = this.createConfigureIntelliSenseStatusBarItem();
         void this.ShowConfigureIntelliSenseButton(false, this.currentClient);
+    }
 
-        this.intelliSenseStatusItem = vscode.languages.createLanguageStatusItem(`cpptools.status.${LanguageStatusPriority.Mid}.intellisense`, util.documentSelector);
-        this.intelliSenseStatusItem.name = localize("cpptools.status.intellisense", "C/C++ IntelliSense Status");
-        this.intelliSenseStatusItem.text = this.idleIntelliSenseText;
+    //#region IntelliSense language status
```

Comment:
Is this supposed to be commented out?

Reply:
That's the syntax for regions in Typescript. By typing "#region" it automatically puts "//".